File: src/Web/Desnz.Chmm.Web/ClientApp/src/app/stores/scheme-year-configuration/commands/update-scheme-year-configuration.command.ts
export interface UpdateSchemeYearConfigurationCommand {
  schemeYearId: string;
  percentageCap: number;
  targetMultiplier: number;
  gasBoilerSalesThreshold: number;
  oilBoilerSalesThreshold: number;
  targetRate: number;
  creditCarryOverPercentage: number;
  alternativeRenewableSystemFuelTypeWeightingValue: number;
  alternativeFossilFuelSystemFuelTypeWeightingValue: number;
}

File: src/Api/Configuration/Desnz.Chmm.Configuration.Common/Commands/UpdateSchemeYearConfigurationCommand.cs
using MediatR;
using Microsoft.AspNetCore.Mvc;

namespace Desnz.Chmm.Configuration.Common.Commands;

public class UpdateSchemeYearConfigurationCommand : IRequest<ActionResult>
{
    public Guid SchemeYearId { get; init; }
    public decimal TargetRate { get; init; }
    public decimal PercentageCap { get; init; }
    public decimal TargetMultiplier { get; init; }
    public decimal CreditCarryOverPercentage { get; init; }
    public int GasBoilerSalesThreshold { get; init; }
    public int OilBoilerSalesThreshold { get; init; }
    public decimal AlternativeRenewableSystemFuelTypeWeightingValue { get; init; }
    public decimal AlternativeFossilFuelSystemFuelTypeWeightingValue { get; init; }
}
